apps/payment/models.py

```diff
@@ -298,9 +302,110 @@ def save(self, *args, **kwargs):
                 raise NotAcceptable("Insufficient funds")
 
             self.user.save()
-        super(PaymentTransaction, self).save(*args, **kwargs)
+        super().save(*args, **kwargs)
+        receipt = PaymentReceipt(object_id=self.id,
+                                 content_type=ContentType.objects.get_for_model(self))
+        receipt.save()
 
     class Meta:
         ordering = ['-datetime']
         verbose_name = _('transaksjon')
         verbose_name_plural = _('transaksjoner')
+
+
+class PaymentReceipt(models.Model):
+    """Transaction receipt"""
+    receipt_id = models.UUIDField(default=uuid.uuid4)
+    content_type = models.ForeignKey(ContentType, on_delete=models.CASCADE)
+    """Which model the receipt is created for"""
+    object_id = models.PositiveIntegerField(null=True)
+    """Object id for the model chosen in content_type"""
+    content_object = GenericForeignKey()
+    """Helper attribute which points to the object select in object_id"""
+
+    def save(self, *args, **kwargs):
+        # Saldo
+        if self._is_type(PaymentTransaction):
```

Comment:
The logic in these if/else statements could be pulled out to different functions, so you do
```
if type == PaymentTransaction:
    get_payment_transaction_items(self)
elif type == PaymentRelaton:
    get_payment_relation_items(self)
```
instead of this huge mess :zipper_mouth_face: :smile: 

Another idea is to make each payment type of object expose `get_title`, `get_timestamp`, `get_description`, … so you can get each of the properties by doing `title = self.get_title()` instead of the whole if/else thingymajig.